### DISCUSSION TITLE
Create new entity based on math operation

### DISCUSSION THREAD

**Discussion by ganzziani** [emporia-vue-local/esphome#296] [Q&A] [answered]
I've got a Vue 3 and I've used all 16 inputs on my breaker box, I am monitoring the most important areas and devices in my home, but I'm not monitoring the A/C power. I figured that the Total Power minus the power on the other 16 signals will be a close enough approximation of the A/C Power. How can I create a new Power entity that is the Difference of the Total Power minus the sum of the 16 channels? And, same thing for the Energy.

**Comment by ganzziani** [comment, marked as answer]
This seems to be working: to get the approximation of the AC power:

```
 - platform: template
    name: "Power AC"
    id: ACPower
    unit_of_measurement: "W"
    update_interval: 10s
    lambda: return id(total_power).state-id(cir1).state-id(cir2).state-id(cir3).state-id(cir4).state-id(cir5).state-id(cir6).state-id(cir7).state-id(cir8).state-id(cir9).state-id(cir10).state-id(cir11).state-id(cir12).state-id(cir13).state-id(cir14).state-id(cir15).state-id(cir16).state;
```
And the for the AC energy, I added one line under  - platform: total_daily_energy

`- { power_id: ACPower, platform: total_daily_energy, accuracy_decimals: 0, name: "Energy AC Power" }`

**Comment by gekkehenkie11** [comment]
This wont help your specific situation, since you can't subtract, but in general for "Create new entity based on math operation" take a look at "helpers". Go to settings/devices&services. Choose "helpers" right top of the screen, then "Create helper" and then "Combine the state of several sensors". Now you can just add the sensors you would like and then choose a 'statistic characteristic' like "Sum".

**Comment by ganzziani** [comment]
How is it that I cannot subract?
I understand that using lambda is similar to writing code, so, I can do any math operation.
The new entity for AC power does provide a good representation of when the AC is on.
And the energy usage helps me visualize that the AC is what consumes most energy at home and it happens when it is hot outside.
It is not perfect since the Total power minus the 16 channels contains not only the AC, but also some other outlets which I am not monitoring.

Below are some picture from today.

![power](https://github.com/user-attachments/assets/5bb74270-074b-42d2-90f9-70677fec42d2)
![energy](https://github.com/user-attachments/assets/e581f228-e1f7-44d9-8656-6d48dbef9f20)

**Reply by gekkehenkie11** [reply]
> How is it that I cannot subract?

Haha you misread my message. You can substract the way you did it (which is the way that's given in the sample YAML file on this website too). But with 'helpers' you can't do that. But if there's ever a situation where you just want to add several sensors, you might want to take a look at 'helpers' like I explained above, they're very easy an quick to use.
